evalml/tests/automl_tests/test_automl.py

```diff
@@ -2086,3 +2086,35 @@ def test_timeseries_baseline_init_with_correct_gap_max_delay(mock_fit, mock_scor
     # Best pipeline is baseline pipeline because we only run one iteration
     assert automl.best_pipeline.parameters == {"pipeline": {"gap": 6, "max_delay": 3},
                                                "Time Series Baseline Regressor": {"gap": 6, "max_delay": 3}}
+
+
+@pytest.mark.parametrize('problem_type', [ProblemTypes.BINARY, ProblemTypes.MULTICLASS,
+                                          ProblemTypes.TIME_SERIES_REGRESSION, ProblemTypes.REGRESSION])
+def test_automl_does_not_include_positive_only_objectives_by_default(problem_type, X_y_regression):
+
+    X, y = X_y_regression
+
+    only_positive = []
+    for name in get_all_objective_names():
+        objective_class = get_objective(name)
+        if objective_class.positive_only:
+            only_positive.append(objective_class)
+
+    search = AutoMLSearch(X_train=X, y_train=y, problem_type=problem_type,
+                          problem_configuration={'gap': 0, 'max_delay': 0})
+    assert search.objective not in only_positive
+    assert all([obj not in only_positive for obj in search.additional_objectives])
+
+
+def test_automl_validate_objective(X_y_regression):
```

Comment:
I don't think we had coverage for this before? Let me know if I'm being redundant!

Reply:
@freddyaboulton Good addition, but why specifically for these 2 objectives?

Reply:
Just cause they were non-core but I changed the test to check all non-core objectives hehe!